.github/workflows/changed_file_trigger_examples_check.yml

```diff
@@ -0,0 +1,117 @@
+name: change_examples_check_all
+run-name: ${{ github.actor }} is testing out GitHub Actions 🚀
+on:
+  pull_request:
+
+  schedule:
+    # run at 00:00 of every Sunday
+    - cron:  '0 0 * * *'
+
+jobs:
+
+# This is for changed files detect and check.
+  detect-changed-files:
+    runs-on: ubuntu-latest
+    outputs:
+      changedFiles: ${{ steps.changed-files.outputs.all_changed_files }}
+
+    name: Check out all files
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+
+      - name: Get all changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          since_last_remote_commit: true  # Using this can trigger action each time a PR is submitted.
+
+
+  deploy:
+    name: Test each changed example files
+    needs: detect-changed-files
+    runs-on: self-hosted
+    container:
+      image: hpcaitech/pytorch-cuda:1.11.0-11.3.0
+      options: --gpus all --rm -v /data/scratch/cifar-10:/data/scratch/cifar-10
```

Comment:
The volume mounting is no longer suitable for this workflow. There are two ways to deal with datasets.

1. The example should natively support test with synthetic data
2. All datasets used for example testing are stored in a specific directory and that directory is mounted by Docker.